cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/SpeakerEmotionInference.scala

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.cognitive
+
+import com.microsoft.azure.synapse.ml.cognitive.TextToSpeechSSMLFormat.TextToSpeechSSMLResponseFormat
+import com.microsoft.azure.synapse.ml.logging.BasicLogging
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import com.microsoft.azure.synapse.ml.stages.Lambda
+import org.apache.http.client.methods.HttpRequestBase
+import org.apache.http.entity.{AbstractHttpEntity, StringEntity}
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.ml.{ComplexParamsReadable, NamespaceInjections, PipelineModel, Transformer}
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.types.StructType
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json._
+
+trait HasLocaleCol extends HasServiceParams {
+  val locale = new ServiceParam[String](this,
+    "locale",
+    s"The locale of the input text",
+    isRequired = true)
+
+  def setLocale(v: String): this.type = setScalarParam(locale, v)
+
+  def setLocaleCol(v: String): this.type = setVectorParam(locale, v)
+}
+
+trait HasVoiceCol extends HasServiceParams {
+  val voice = new ServiceParam[String](this,
+    "voice",
+    s"The name of the voice used for synthesis",
+    isRequired = true)
+
+  def setVoiceName(v: String): this.type = setScalarParam(voice, v)
+
+  def setVoiceNameCol(v: String): this.type = setVectorParam(voice, v)
+}
+
+trait HasInput extends HasServiceParams {
+  val input = new ServiceParam[String](this,
+    "input",
+    s"The text input to annotate with inferred emotion")
+
+  def setInput(v: String): this.type = setScalarParam(input, v)
+
+  def setInputCol(v: String): this.type = setVectorParam(input, v)
+}
+
+trait HasSpeakerEmotionInferenceParams extends HasServiceParams
+  with HasLocaleCol with HasVoiceCol with HasSetLocation
+  with HasInput
+
+object SpeakerEmotionInference
+  extends ComplexParamsReadable[SpeakerEmotionInference] with Serializable
+
+class SpeakerEmotionInference(override val uid: String) extends CognitiveServicesBase(uid)
+  with HasSpeakerEmotionInferenceParams with HasCognitiveServiceInput
+  with HasInternalStringOutputParser
+  with BasicLogging {
+  logClass()
+
+  def this() = this(Identifiable.randomUID(classOf[SpeakerEmotionInference].getSimpleName))
+
+  setDefault(
+    locale -> Left("en-US"),
+    voice -> Left("en-US-SaraNeural"),
+    input -> Left(this.uid + "_input"),
+  )
+
+  def urlPath: String = "cognitiveservices/v1"
+
+  protected val additionalHeaders: Map[String, String] = Map[String, String](
+    ("X-Microsoft-OutputFormat", "textanalytics-json"),
+    ("Content-Type", "application/ssml+xml")
+  )
+
+  override def inputFunc(schema: StructType): Row => Option[HttpRequestBase] = super.inputFunc(schema).andThen(r => {
```

Comment:
i think you might be able to simplify this with .map insteead of .andThen, but i didnt know about andThen!

Reply:
As discussed, leaving as andThen since we're dealing with a curried func instead of a raw option result